### ISSUE TITLE
Using a dynamic template only sends the text/plain part

### ISSUE THREAD

**Issue by koppen** [eddiezane/sendgrid-actionmailer#68]
We use the dynamic template feature of SendGrid quite heavily. All of our templates contain both a HTML version and a Plain Text version (both auto generated and manually created). This has worked fine until august 13th or so, after which our emails stopped containing the text/html part.

## How to reproduce

Given a Dynamic Template with id `d-8bc581e6c95d4a9a8f47ee85bcda656a` containing both a HTML and a Plain Text version, we set up a mailer with the following method:

```ruby
class SendgridMailer < ApplicationMailer
  def test
    mail(
      :to => "recipient@example.com",
      :body => "Test email body",
      :template_id => "d-8bc581e6c95d4a9a8f47ee85bcda656a"
    )
  end
end

SendgridMailer.test.deliver_now
```

## Actual result

- I receive an email with only a `text/plain` part.

## Expected

- I receive an email with both a `text/html` and a `text/plain` part.

## Details

After august 13th our emails started containing only a `text/plain` part, ie the expected `text/html` part was nowhere to be seen. After talking to SendGrid support it was determined that we should stop sending a `content_type` in the request to them and SendGrid would figure out what parts to include.

However, that is proving tricky using sendgrid-actionmailer.

Thanks to ActionMailer a `:body` key is required, and following the instructions for [using `template_id`](https://github.com/eddiezane/sendgrid-actionmailer#template_id-string) it is suggested to just set it even though it isn't used.

Sending an email using the following arguments:

```ruby
mail(
  :to => "recipient@example.com",
  :body => "Test email body",
  :template_id => "d-8bc581e6c95d4a9a8f47ee85bcda656a"
)
```

This does deliver an email, but the email contains only a `text/plain` part. The JSON generated and submitted to the backend API is:

```json
{
  "from":{
    "email":"sender@example.com",
  },
  "subject":"Test",
  "personalizations":
    [
      {
        "to":[
          {
            "email":"recipient@example.com"
          }
        ]
      }
    ],
    "content":[
      {
        "type":"text/plain",
        "value":"Test email body"
      }
    ],
    "template_id":"d-8bc581e6c95d4a9a8f47ee85bcda656a"
  }
```

As it can be seen, the unused `"Test email body"` is submitted under the key `content.value` and a `content.type` key is added with the value `text/plain`.

If we remove the `content` key and its children and submit the following JSON to the API manually, the email arrives just fine with both a `text/plain` part and `text/html` part as expected:

```json
{
  "from":{
    "email":"sender@example.com",
  },
  "subject":"Test",
  "personalizations":[
    {
      "to":[
        {"email":"recipient@example.com"}
      ]
    }
  ],
  "template_id":"d-8bc581e6c95d4a9a8f47ee85bcda656a"
}
```

Interestingly we get the same effect if we add both wanted MIME types to the `content` key:

```json
{
  "from":{
    "email":"sender@example.com",
  },
  "subject":"Test",
  "personalizations":[
    {
      "to":[
        {"email":"recipient@example.com"}
      ]
    }
  ],
  "content":[
    {
      "type":"text/plain",
      "value":"Test email body"
    },
    {
      "type":"text/html",
      "value":"Test email body"
    }
  ],
  "template_id":"d-8bc581e6c95d4a9a8f47ee85bcda656a"
}
```

## Suggestions

1. We could stop sending the `content` elements if the email also contains `template_id`?
2. We could start including the unused `body` value in both `text/plain` and `text/html` `content` parts when the email contains `template_id`.

Personally, I am leaning towards the first option, as that is the simpler payload transmitted over the wire, it makes no assumptions about what the sender wants, and it's what SendGrid explained to us should happen.

However, option 2 is closer to what the payload is likely to look like when not using `template_id` and is more explicit.

**Comment by tyrauber** [comment]
@koppen Thank you for the detailed Issue and Pull Request.   I wonder if the issue started with [the update to sendgrid-ruby ~> 6 on July 9th](https://github.com/eddiezane/sendgrid-actionmailer/commit/5f1124d6b39e5a160e7ee50971745e34de5570c0) or whether it was a Sendgrid API change.

Regardless, I agree with your assessment.  I think my preference would be option 1. Remove the content from the payload if a template_id is provided.  Since the content will be ignored by Sendgrid if a template_id is supplied, and a body is only required because Mail raises an error without one,  I think removing it from the payload is appropriate.

**Comment by koppen** [comment]
I am pretty sure this is a SendGrid API change. I spent a good amount of time with their support and it sounded like we weren't the only ones that hit this issue; in any case they knew exactly what could be the fix without knowing what library we used.

Also, the behavior is consistent when using the API directly via curl, ie not using sendgrid-ruby.

> Regardless, I agree with your assessment. I think my preference would be option 1. Remove the content from the payload if a template_id is provided. Since the content will be ignored by Sendgrid if a template_id is supplied, and a body is only required because Mail raises an error without one, I think removing it from the payload is appropriate.

Phew, that's what I did 😅

**Comment by tyrauber** [comment]
:+1:  Merged and bumped to 2.6.0.  Thanks for the Pull Request!

**Comment by koppen** [comment]
This is part of what SendGrid support told us when we talked to them, again reinforcing it is a behavior change on their end:

> Have you tried to remove any content header indicating plain text? We recently removed our plain text feature, which means if you are specifying a content header in your request body it will take precedent.
